dali/sasha/saxref.cpp

```diff
@@ -903,6 +940,9 @@ class CNewXRefManager: public CNewXRefManagerBase
             return false;
         }
         StringBuffer fname;
+        unsigned numDevices = 0;
+        if (pdir == root && !storagePlane->hasProp("@HostGroups"))
+            unsigned numDevices = storagePlane->getPropInt("@numDevices");
```

Comment:
slightly surprised this didn't cause a compile error, unintended stack local 'unsigned'

Reply:
numDevices here and on line 943 should default to 1. Striped if >1

Reply:
you may have used 0 and >0, because constructPartFilename is adding a strip dir even if 1?
constructPartFilename should skip calcStripeNum if numDevices <= 1, and pass 0 into makePhysicalPartName

Reply:
also, it's going to be doing this for every directory as it recurses..
Better to move out, do once, and store result in a "isPlaneStriped" member for efficiency/clarity.